src/prefect/cli/build_register.py

```diff
@@ -0,0 +1,972 @@
+    cache = dict.fromkeys(path_to_module)
+    while True:
+        cache2 = {}
+        for path in tracked:
+            try:
+                try:
+                    with os.scandir(path) as directory:
+                        for entry in directory:
+                            if entry.is_file() and entry.path.endswith(".py"):
+                                old_mtime = cache.get(entry.path)
+                                mtime = entry.stat().st_mtime
+                                if mtime != old_mtime:
+                                    cache2[entry.path] = mtime
+                except NotADirectoryError:
+                    old_mtime = cache.get(path)
+                    mtime = os.stat(path).st_mtime
+                    if mtime != old_mtime:
+                        cache2[path] = mtime
+            except FileNotFoundError:
+                cache.pop(path, None)
+
+        if cache2:
+            change_paths = []
+            change_mods = []
+            for path in cache2:
+                module = path_to_module.get(path)
+                if module is not None:
+                    change_mods.append(module)
+                else:
+                    change_paths.append(path)
+            if change_paths or change_mods:
+                yield change_paths, change_mods
+                cache.update(cache2)
+
+        time.sleep(period)
+
+
+REGISTER_EPILOG = """
+\bExamples:
+
+\b  Register all flows found in a directory.
+
+\b    $ prefect register --project my-project -p myflows/
+
+\b  Register a flow named "example" found in `flow.py`.
+
+\b    $ prefect register --project my-project -p flow.py -n "example"
+
+\b  Register all flows found in a module named `myproject.flows`.
+
+\b    $ prefect register --project my-project -m "myproject.flows"
+
+\b  Register all pre-built flows from a remote JSON file.
+
+\b    $ prefect register --project my-project --json https://some-url/flows.json
+
+\b  Watch a directory of flows for changes, and re-register flows upon change.
+
+\b    $ prefect register --project my-project -p myflows/ --watch
+"""
+
+
+@click.group(invoke_without_command=True, epilog=REGISTER_EPILOG)
+@click.option(
+    "--project",
+    help="The name of the Prefect project to register this flow in. Required.",
+    default=None,
+)
+@click.option(
+    "--path",
+    "-p",
+    "paths",
+    help=(
+        "A path to a file or a directory containing the flow(s) to register. "
+        "May be passed multiple times to specify multiple paths to register."
+    ),
+    multiple=True,
+)
+@click.option(
+    "--module",
+    "-m",
+    "modules",
+    help=(
+        "A python module name containing the flow(s) to register. May be "
+        "passed multiple times to specify multiple modules to register."
+    ),
+    multiple=True,
+)
+@click.option(
+    "--json",
+    "-j",
+    "json_paths",
+    help=(
+        "A path or URL to a json file containing the flow(s) to register. "
```

Comment:
The fact that the schema of this file is not a dump of `flow.serialize()` took me a minute to work through and seems like something we should call out here -- not sure of the simplest way to say "we expect this file to be the output of `prefect build`

Reply:
Sure, I can expand the help string here.

Reply:
Done